icu4c/source/common/unicode/localpointer.h

```diff
@@ -42,7 +42,11 @@
 
 #if U_SHOW_CPLUSPLUS_API
 
+#ifndef U_IN_DOXYGEN
```

Comment:
This ^ is required since the U_SHOW_CPLUSPLUS_API is forced by doxygen without ensuring C++, I saw this failing in the CI, not sure how it has worked so far.

Reply:
because Doxyfile can set a specific value for the macro. I'd rather not add `U_IN_DOXYGEN` if not needed.